recipes/apriltag/all/conanfile.py

```diff
@@ -0,0 +1,53 @@
+import os
+import stat
+import shutil
+from conans import ConanFile, tools, CMake, AutoToolsBuildEnvironment
+from conans.errors import ConanException
+
+class apriltagConan(ConanFile):
+    name = "apriltag"
+    license = "BSD 2-Clause"
+    homepage = "https://april.eecs.umich.edu/software/apriltag"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("AprilTag is a visual fiducial system, useful for a wide variety of tasks \
+                    including augmented reality, robotics, and camera calibration")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, "fPIC": True}
+    generators = "cmake"
+    topics = ("conan", "apriltag", "robotics")
+    exports_sources = ["CMakeLists.txt", "*.patch"]
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
Patches should be applied in the `source()` step, not in `build()`

Reply:
I prefer keeping it in `build()`, because the source folder is the raw copy of the project. In the build folder we can edit/remove, do anything there, but when we need to "debug" a recipe, keeping the source folder clean for inspection or just checking the code before applying a patch is better.

Reply:
I just moved the patching into source(), but I also think that doing it inside build() is cleaner. The conan guidelines do not specify where the patching should be done.

Reply:
I moved the patches back to the build() step.